frontend/src/components/EditItem.tsx

```diff
@@ -0,0 +1,134 @@
+import {useEffect, useState} from "react";
+import {Todo} from "../model";
+import "./EditItem.css"
+
+interface EditItemProps {
+    fetchAll: () => void;
+    editMode: string;
+    setEditMode: (input: string) => void;
+    editId: string;
+}
+
+export default function EditItem (props: EditItemProps) {
+
+    const [task, setTask] = useState<string>("");
+    const [description, setDescription] = useState<string>("");
+    const [status, setStatus] = useState<string>("");
+
+    useEffect(()=>{
+        if(props.editMode === "edit"){
+            getTaskById();
+        }
+    }, [props])
+
+    const getTaskById = () => {
+        console.log(`fetch todo by id`);
+        fetch(`http://localhost:8080/api/kanban/${props.editId}`)
+            .then(response => response.json())
+            .then((todo: Todo) => {
+                setTask(todo.task);
+                setDescription(todo.description);
+                setStatus(todo.status);
+                console.log(todo);
+            });
+    }
+
+    const addNewTodo = () => {
+        console.log(`saving: ${task} ${description}`)
+
+        fetch('http://localhost:8080/api/kanban', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({task: task, description: description, status: "OPEN"}),
+        })
+            .then(response => response.json())
+            .then(data => {
+                console.log('Success:', data);
+                props.fetchAll();
+                props.setEditMode("view");
```

Comment:
Du wechselst hier den `editMode` bevor die `fetchAll`-Response da ist.
Du könntest folgendes tun:
1. Den Typ von `fetchAll` folgendermaßen definieren: `fetchAll: () => Promise<any>`. Also erwarten, dass `fetchAll` ein `Promise` zurückgibt.
2. An dieser Stelle hier kannst du mit dem `.then(() => props.setEditMode("view"))` dafür sorgen, dass `props.fetchAll()` sicher durch ist, bevor die view gewechselt wird.
3. In `getAllTodos` ein `return` vor das `fetch` schreiben.

Reply:
Ich glaube es müsste auch so gehen:
```typescript
.then(response => response.json())
.then(data => props.fetchAll())
.then(() => props.setEditMode("view"));
```